bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/handler/HydrawiseAccountHandler.java

```diff
@@ -184,20 +187,40 @@ private synchronized void initPolling(int initalDelay, int refresh) {
         pollFuture = scheduler.scheduleWithFixedDelay(this::poll, initalDelay, refresh, TimeUnit.SECONDS);
     }
 
+    /**
+     * The API will randomly reject a request with a 401 not authorized, waiting a min and refreshing the token usually
+     * fixes it
+     */
+    private synchronized void retryToken() {
+        clearTokenRefresh();
+        tokenFuture = scheduler.schedule(() -> {
+            try {
+                OAuthClientService oAuthService = this.oAuthService;
+                if (oAuthService != null) {
+                    oAuthService.refreshToken();
+                    initPolling(0, MIN_REFRESH_SECONDS);
+                }
+            } catch (OAuthException | IOException | OAuthResponseException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        }, TOKEN_REFRESH_SECONDS, TimeUnit.SECONDS);
+    }
+
     /**
      * Stops/clears this thing's polling future
      */
     private void clearPolling() {
-        ScheduledFuture<?> localFuture = pollFuture;
-        if (isFutureValid(localFuture)) {
-            if (localFuture != null) {
-                localFuture.cancel(false);
-            }
-        }
+        clearFuture(pollFuture);
     }
 
-    private boolean isFutureValid(@Nullable ScheduledFuture<?> future) {
-        return future != null && !future.isCancelled();
+    private void clearTokenRefresh() {
+        clearFuture(tokenFuture);
+    }
+
+    private void clearFuture(@Nullable final ScheduledFuture<?> future) {
+        if (future != null && !future.isCancelled()) {
+            future.cancel(false);
```

Comment:
When this is called by `dispose` it should be:
```suggestion
        if (future != null) {
            future.cancel(true);
```

Would this also be working for `retryToken` scenario?

Reply:
Thanks, and yes this will work for both.